pkg/controller/cronjob/cronjob_controllerv2.go

```diff
@@ -739,3 +738,31 @@ func deleteJob(cj *batchv1.CronJob, job *batchv1.Job, jc jobControlInterface, re
 func getRef(object runtime.Object) (*corev1.ObjectReference, error) {
 	return ref.GetReference(scheme.Scheme, object)
 }
+
+func formatSchedule(cj *batchv1.CronJob, recorder record.EventRecorder) string {
+	timeZoneEnabled := utilfeature.DefaultFeatureGate.Enabled(features.CronJobTimeZone)
+	scheduleContainsTZ := false
+
+	if strings.Contains(cj.Spec.Schedule, "TZ") || strings.Contains(cj.Spec.Schedule, "CRON_TZ") {
+		scheduleContainsTZ = true
+		recorder.Eventf(cj, corev1.EventTypeWarning, "UnsupportedSchedule", "CRON_TZ or TZ used in schedule %q is not officially supported, see https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/ for more details", cj.Spec.Schedule)
+	}
+
+	if timeZoneEnabled && len(cj.Spec.TimeZone) > 0 {
+		if scheduleContainsTZ {
+			fields := strings.SplitN(cj.Spec.Schedule, " ", 2)
+			if len(fields) == 2 {
+				cj.Spec.Schedule = fields[1]
```

Comment:
the TZ/CRON_TZ check was just Contains, not HasPrefix; there's no guarantee there weren't leading spaces or other things in front of TZ/CRON_TZ

Reply:
also, this is mutating the input, which is surprising (and not allowed from some callsites like updateCronJob())

Reply:
Good catch, @deejross :arrow_up: